module/wa/module.py

```diff
@@ -967,18 +968,44 @@ def import_wa(i):
             pname=None
             imported_params={}
 
-            for name, obj in inspect.getmembers(cs):
-                if name!='AndroidUxPerfWorkload' and inspect.isclass(obj):
-                    try:
-                        addr=getattr(cs, name)
-                        pp=addr.parameters
-                        pname=addr.package
-                    except Exception as e:
-                        pass
-
-                    if pp!=None:
-                        wa_name=name
-                        break
+            # Only examine classes
+            for name, obj in inspect.getmembers(cs, predicate=inspect.isclass):
```

Comment:
How CK runs WA is orthogonal to this. Nothing should need to be changed elsewhere in the code. You would have to import `ExtensionLoader`  from `wlauto`, which can be done inside an import guard, as CK can't assume WA is installed, but then the line above could be changed to

```
try:
    from wlauto import ExtensionLoader
except ImportError:
    ExtensionLoader = None

if ExtensionLoader:
     for wkload_cls in ExtensionLoader().list_workloads():
         # do whatever with  wkload_cls

```
without the need for any of the inspection stuff. This would also be more technically correct, since WA workloads will not necessarily all be within the `wlauto` package.